### ISSUE TITLE
sidecar container inject failed on kubernetes 1.14.3

### ISSUE THREAD

**Issue by like-inspur** [tumblr/k8s-sidecar-injector#42]
I use k8s-sidecar-injector 0.1.8 before on kubernetes 1.14.3 and found sidecar container inject suceessfully like below:
```
I0521 03:55:42.351044       1 webhook.go:480] AdmissionResponse: patch=[{"op":"add","path":"/spec/containers/-","value":{"name":"exporter","image":"docker.inspur.com:5000/service/lma/consul-exporter:0.6.0","args":["--consul.server=localhost:8500","--log.level=warn"],"ports":[{"name":"metric","containerPort":9107}],"resources":{},"livenessProbe":{"tcpSocket":{"port":"metric"},"initialDelaySeconds":60,"timeoutSeconds":10,"periodSeconds":60},"readinessProbe":{"tcpSocket":{"port":"metric"},"initialDelaySeconds":60,"timeoutSeconds":10,"periodSeconds":60},"imagePullPolicy":"IfNotPresent"}},{"op":"add","path":"/metadata/annotations/injector.inspur.com~1status","value":"injected"}]
```

But when I upgrade k8s-sidecar-injector to 0.4.0 and found sidecar container inject failed like below:
```
I0521 06:20:35.624586       1 webhook.go:468] AdmissionReview for Kind=/v1, Kind=Pod, Namespace=monitoring Name= (consul-0) UID=2e9e0283-9b2b-11ea-9df2-5254009197f9 patchOperation=CREATE UserInfo={system:serviceaccount:kube-system:statefulset-controller c7bb22c8-6cef-11ea-9df2-5254009197f9 [system:serviceaccounts system:serviceaccounts:kube-system system:authenticated] map[]}
I0521 06:20:35.624673       1 webhook.go:165] Pod monitoring/consul-0 annotation injector.tumblr.com/request is missing, skipping injection
I0521 06:20:35.624688       1 webhook.go:474] Skipping mutation of monitoring/consul-0: Missing injection request annotation
I0521 06:20:35.624733       1 webhook.go:584] Ready to write reponse ...
```
The config of sidecar container is like below and not be modified after k8s-sidecar-injector upgrade
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: sidecar-exporter-consul
  labels:
    app: k8s-sidecar-injector
data:
  sidecar-exporter-consul: |
    name: sidecar-exporter-consul
    containers:
    - name: exporter
      image: docker.inspur.com:5000/service/lma/consul-exporter:0.6.0
      imagePullPolicy: {{ .Values.image.pullPolicy }}
      args:
      - --consul.server=localhost:8500
      - --log.level=warn
      ports:
      - containerPort: 9107
        name: metric
```
The config of pod is like below and has request annotation, but sidecar container inject failed
```
apiVersion: v1
kind: Pod
metadata:
  annotations:
    injector.inspur.com/request: sidecar-exporter-consul
    prometheus.io/port: "9107"
    prometheus.io/scrape: "true"
```

**Comment by byxorna** [comment]
Hey @like-inspur - looks like the injector decided that the annotation required is `injector.tumblr.com/request`.

From your logs:
```
I0521 06:20:35.624673       1 webhook.go:165] Pod monitoring/consul-0 annotation **injector.tumblr.com/request is missing**, skipping injection
```

https://github.com/tumblr/k8s-sidecar-injector/blob/master/cmd/main.go#L63 looks like you can pass the `--annotation-namespace` argument to the service at boot to change that.

**Comment by like-inspur** [comment]
I have config env ANNOTATION_NAMESPACE as injector.inspur.com in k8s-sidecar-injector deployment and work fine with 0.1.8 version, but work failed with 0.4.0

**Comment by byxorna** [comment]
> I have config env ANNOTATION_NAMESPACE as injector.inspur.com in k8s-sidecar-injector deployment and work fine with 0.1.8 version, but work failed with 0.4.0

Interesting - I did a quick grep back through the 0.1.8 release (and HEAD) and cannot find anywhere we are pulling the ANNOTATION_NAMESPACE out of the environment. This is very curious that it was working for you before!

The fix here would be to pass `--annotation-namespace=injector.inspur.com` to the entrypoint script. Sorry about that odd behavior! If you would like to submit a PR to handle `ANNOTATION_NAMESPACE` in `entrypoint.sh`, it would be much appreciated!

**Comment by like-inspur** [comment]
sorry， I forget I have modify dockerfile on 0.1.8 like you previously mentioned, I'd like to create pull request for this

**Comment by byxorna** [comment]
Fixed via #31. The docker image in `tumblr/k8s-sidecar-injector:latest` will respect `ANNOTATION_NAMESPACE` env var now :)

Thanks for reporting this!
